### PR TITLE
fix: file upload size limit and API error response

### PR DESCRIPTION
**Pull request by milmigi**
This pull request introduces several enhancements and new features to the `llassist` API and web services, primarily focusing on file upload settings and error handling improvements. The most important changes include adding a new search endpoint in the `AppSettingsController`, integrating file upload settings into the `ProjectController`, and implementing a comprehensive file validation mechanism.

### Enhancements to API Controllers:
* [`llassist.ApiService/Controllers/AppSettingsController.cs`](diffhunk://#diff-8753a26ea82c068a9b8d0b551ec69415ed6f132d0b4d9d6bd5e16ab54ca1cf9bR59-R65): Added a `Search` endpoint to allow searching for app settings based on specified criteria.
* [`llassist.ApiService/Controllers/ProjectController.cs`](diffhunk://#diff-f46891d8b870dd2ba2a9c6d610793ddbffe8a36f4304cb6ad0c0ce96219031e1L18-R33): Integrated `IAppSettingService` to manage file upload settings and added validation logic for file uploads. [[1]](diffhunk://#diff-f46891d8b870dd2ba2a9c6d610793ddbffe8a36f4304cb6ad0c0ce96219031e1L18-R33) [[2]](diffhunk://#diff-f46891d8b870dd2ba2a9c6d610793ddbffe8a36f4304cb6ad0c0ce96219031e1R95-R129)

### Introduction of File Upload Settings:
* [`llassist.Common/Models/FileUploadSettings.cs`](diffhunk://#diff-0d35662267b19d234371549a1983f839eca694e4f35aabf7a4fbeff293fbdef9R1-R45): Created a `FileUploadSettings` class to encapsulate file upload configurations, including maximum file size and allowed extensions.
* [`llassist.Common/Validators/FileValidator.cs`](diffhunk://#diff-687e22c803e31b495f2334a55663d20e56c099179116a6cbc31052bb8d5028f7R1-R44): Implemented `FileValidator` to validate file uploads against the configured settings.

### Improved Error Handling:
* [`llassist.ApiService/Program.cs`](diffhunk://#diff-40818100b5106996df3cd89724d4b97d5952ef96ec982bf947a950acb2b4db28L98-R110): Replaced `ProblemDetails` with `ApiErrorResponse` for a more detailed error response structure.
* [`llassist.Common/Models/Responses/ApiErrorResponse.cs`](diffhunk://#diff-793805d302ec97cd109b37026eeb3bf0d8b00115de1b54211a5bc3f9e8e60ad6R1-R9): Defined the `ApiErrorResponse` class to standardize error responses.

### Additional Changes:
* [`llassist.Common/Models/AppSetting.cs`](diffhunk://#diff-ef45e6cdea1679c42e3c73f15ae6d3393f263dc025a05274d046b86ab2faead2R11-R29): Added `AppSettingKeys` to centralize app setting keys.
* [`llassist.Web/Components/Pages/Projects/ProjectList.razor`](diffhunk://#diff-dbda77b09957a8aef16caa4c439ae1882799f27f8e907b61b6b5d68df6e85830R64-R88): Updated the project list component to display file upload settings and handle file uploads more robustly. [[1]](diffhunk://#diff-dbda77b09957a8aef16caa4c439ae1882799f27f8e907b61b6b5d68df6e85830R64-R88) [[2]](diffhunk://#diff-dbda77b09957a8aef16caa4c439ae1882799f27f8e907b61b6b5d68df6e85830R236-R259)

### CODE REVIEW

**Review by cyharyanto — COMMENTED** [review]
What do you think about where to put the checking? Should it be part of the ApiService instead of the Web?

Overall:

**ApiService:**
👍 Uses clean error response model (`ApiErrorResponse`) with timestamp and details
❓ No file validation implementation yet in the backend

**Web:**
👍 Multiple validation layers:
  - Frontend checks (`IsValidUpload` method) before upload starts
  - Settings injected via DI (`FileUploadSettings`)
  - Visual feedback (`uploadError`, `isUploading states`)
  - Max size enforced at stream level (`OpenReadStream` with `maxAllowedSize`)

❓ Yet, duplicate configuration needed in both `FormOptions` and `FileUploadSettings`

**Both Components:**
👍 Central configuration via `appsettings.json`
👍 Type-safe settings via `FileUploadSettings` class
❓ Risk of mismatched validation rules if not kept in sync

**Recommendations:**
👍 Common, reusable checking code in `llassist.Common`
👍 `ApiService` to load the configuration, not `Web`
👍 Can consider to use the in-database configuration too

**Comment by milmigi** [comment]
@cyharyanto both layer should have the checking, since we need to explicitly increase the file read stream and server-side frontend also need to be protected. We can move the configuration to backend and open an API for the frontend to retrieve and do validation.

File: llassist.ApiService/Services/FileUploadSettingsService.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        return new FileUploadSettings
+        {
+            MaxSizeMB = maxFileSizeMBSetting != null 
+                ? int.Parse(maxFileSizeMBSetting.Value)
+                : DefaultMaxFileSizeMB,
+
+            AllowedExtensions = allowedExtensionsSetting?.Value?.Split(',') ?? DefaultAllowedExtensions
+        };
+    }
```
Comment (cyharyanto) [inline llassist.ApiService/Services/FileUploadSettingsService.cs:38]: TBH, I think having a specialized service just for `FileUploadSettings` is a bit too scoped.

File: llassist.ApiService/Controllers/FileUploadSettingsController.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    }
+
+    [HttpGet]
+    public async Task<ActionResult<FileUploadSettings>> Get()
+    {
+        var settings = await _fileUploadSettingsService.GetFileUploadSettingsAsync();
+        if (settings == null) return NotFound();
+        return Ok(settings);
+    }
+}
```
Comment (cyharyanto) [inline llassist.ApiService/Controllers/FileUploadSettingsController.cs:25]: I think it is much better just to use the common app settings. I imagine the file upload settings can eventually be configured on the database level.

**Review by cyharyanto — CHANGES_REQUESTED** [review]

File: llassist.ApiService/Services/FileUploadSettingsService.cs (hunk 1 above)
Comment (milmigi) [reply llassist.ApiService/Services/FileUploadSettingsService.cs]: updated

File: llassist.ApiService/Controllers/FileUploadSettingsController.cs (hunk 2 above)
Comment (milmigi) [reply llassist.ApiService/Controllers/FileUploadSettingsController.cs]: as discusses, as of now we're still using common app settings. once we introduce user / owner / tenant, then we have room to configure this settings according to the new hierarchy

**Review by cyharyanto — APPROVED** [review]
Thank you! I think it is much better as in not introducing long-term issue.

**Pull request merged**